### ISSUE TITLE
Multi level autocomplete

### ISSUE THREAD

**Issue by spaja** [ajaxorg/ace#4780]
Hi everyone,

i am trying to implement a multi level autocomplete with Ace.

Imagine this usecase: first level autocomplete is a continent and second level is a country within selected continent.

The problem I am facing is that once the selection is done in the first autocomplete (imagine I selected-clicked on Europe) I want the second autocomplete to open directly (without user typing or using ctrl+space).

I have tried using insertMatch without success. 

This is my current code (which is not opening the second autocomplete directly on the first autocomplete selection):

```javascript
editor.setOptions({enableBasicAutocompletion: true, enableLiveAutocompletion: true, enableSnippets: true});

// uses http://rhymebrain.com/api.html
var rhymeCompleter = {
    getCompletions: function(editor, session, pos, prefix, callback) {
            if (prefix.length === 0) { callback(null, []); return }
            $.getJSON(
                "http://rhymebrain.com/talk?function=getRhymes&word=" + prefix,
                function(wordList) {
                    // wordList like [{"word":"flow","freq":24,"score":300,"flags":"bc","syllables":"1"}]
                    callback(null, wordList.map(function(ea) {
                        return {name: ea.word, value: ea.word, score: ea.score, meta: "rhyme",completer: {
                            insertMatch: function (editor, data) {
                                        editor.completer.insertMatch({value: data.caption});
                            }
                            }}
                    }));
                })
        }
}
langTools.addCompleter(rhymeCompleter);
```

Please let me know if this is possible and if anyone had a similar problem.

Thank you

**Comment by andrewnester** [comment]
Closing as duplicate of https://github.com/ajaxorg/ace/issues/4561
